src/pudl/transform/ferceqr.py

```diff
@@ -0,0 +1,278 @@
+    """
+    dtypes = Resource.from_id(table_name).to_duckdb_dtypes()
+    return persist_table_as_parquet(
+        table_data=table_data.select(
+            *[
+                expression.cast(dtypes[col]).alias(col)
+                for col, expression in col_expressions.items()
+            ],
+        ),
+        table_name=table_name,
+        partitions={"year_quarter": year_quarter},
+    )
+
+
+def _na_to_null(col_name: str) -> duckdb.Expression:
+    """Convert string NA values to NULL."""
+    return duckdb.SQLExpression(
+        f"CASE WHEN UPPER({col_name}) IN ('N/A', 'NA') THEN NULL ELSE UPPER({col_name}) END"
+    )
+
+
+@dg.asset(partitions_def=ferceqr_year_quarters)
+def core_ferceqr__quarterly_identity(
+    context: dg.AssetExecutionContext, raw_ferceqr__ident: ParquetData
+):
+    """Apply data types to EQR ident table."""
+    year_quarter = context.partition_key
+    with duckdb_relation_from_parquet(raw_ferceqr__ident, use_all_partitions=True) as (
+        table,
+        _,
+    ):
+        return transform_eqr_table(
+            table_name="core_ferceqr__quarterly_identity",
+            table_data=table,
+            year_quarter=year_quarter,
+            col_expressions={
+                "year_quarter": duckdb.ColumnExpression("year_quarter"),
+                "company_id_ferc": duckdb.ColumnExpression("company_identifier"),
+                "filer_unique_id": duckdb.ColumnExpression("filer_unique_id"),
+                "company_name": duckdb.ColumnExpression("company_name"),
+                "contact_name": duckdb.ColumnExpression("contact_name"),
+                "contact_title": duckdb.ColumnExpression("contact_title"),
+                "contact_address": duckdb.ColumnExpression("contact_address"),
+                "contact_city": duckdb.ColumnExpression("contact_city"),
+                "contact_state": duckdb.ColumnExpression("contact_state"),
+                "contact_zip": duckdb.ColumnExpression("contact_zip"),
+                "contact_country_name": duckdb.ColumnExpression("contact_country_name"),
+                "contact_phone": duckdb.ColumnExpression("contact_phone"),
+                "contact_email": duckdb.ColumnExpression("contact_email"),
+                "transactions_reported_to_index_price_publishers": (
+                    duckdb.CaseExpression(
+                        condition=duckdb.SQLExpression(
+                            "UPPER(transactions_reported_to_index_price_publishers)"
+                        )
+                        == duckdb.ConstantExpression("Y"),
+                        value=True,
+                    ).when(
+                        condition=duckdb.SQLExpression(
+                            "UPPER(transactions_reported_to_index_price_publishers)"
+                        )
+                        == duckdb.ConstantExpression("N"),
+                        value=False,
+                    )
+                ),
```

Comment:
We have this Y/N to True/False transform in 2 places now (here and down in contracts) so I think we should pull it out into a function. I'm getting type checking errors on the use of `True` and `False` for values. it seems to only want `ConstantExpression`, and I still think the `SQLExpression()` is more concise and readable for these compositions with many parts.

```py
def _yn_to_bool(col: str) -> duckdb.Expression:
    """Convert 'Y' and 'N' values into True and False, respectively. Other values become NULL."""
    return duckdb.SQLExpression(
        f"""
        CASE
            WHEN UPPER({col}) = 'Y' THEN TRUE
            WHEN UPPER({col}) = 'N' THEN FALSE
            ELSE NULL
        END"""
    )
```